rust/src/transformations/make_stable_expr/namespace_str/expr_strptime/test.rs

```diff
@@ -0,0 +1,162 @@
+    )?
+    .lazy();
+
+    let casted = lf.clone().with_columns([
+        col("time").str().strptime(
+            DataType::Time,
+            StrptimeOptions {
+                format: Some("%H:%M:%S".to_string()),
+                strict: false,
+                exact: true,
+                cache: true,
+            },
+            lit("null"),
+        ),
+        col("datetime").str().strptime(
+            DataType::Datetime(TimeUnit::Milliseconds, None),
+            StrptimeOptions {
+                format: Some("%Y-%m-%d %H:%M:%S".to_string()),
+                strict: false,
+                exact: false,
+                cache: true,
+            },
+            lit("null"),
+        ),
+        col("date").str().strptime(
+            DataType::Date,
+            StrptimeOptions {
+                format: Some("%Y-%m-%d".to_string()),
+                strict: false,
+                exact: false,
+                cache: true,
+            },
+            lit("null"),
+        ),
+    ]);
+    let t_casted = make_stable_lazyframe(lf_domain, SymmetricDistance, casted)?;
+
+    let observed = t_casted.invoke(&lf)?.collect()?;
+    let expected = r#"shape: (4, 3)
+┌──────────┬─────────────────────┬──────────────┐
+│ time     ┆ datetime            ┆ date         │
+│ ---      ┆ ---                 ┆ ---          │
+│ time     ┆ datetime[ms]        ┆ date         │
+╞══════════╪═════════════════════╪══════════════╡
+│ 00:00:00 ┆ 2000-01-01 00:00:00 ┆ +10000-01-01 │
+│ null     ┆ null                ┆ null         │
+│ null     ┆ null                ┆ null         │
+│ null     ┆ null                ┆ null         │
+└──────────┴─────────────────────┴──────────────┘"#;
+    assert_eq!(format!("{:?}", observed), expected);
+    Ok(())
+}
+
+#[test]
+fn test_strptime_datetime() -> Fallible<()> {
+    let lf = df!("datetime" => &[
+        "+262142-01-01T00:00:00Z",
+        "+262143-01-01T00:00:00Z",
+        "+262144-01-01T00:00:00Z",
+    ])?
+    .lazy()
+    .with_column(col("datetime").str().strptime(
+        DataType::Datetime(TimeUnit::Milliseconds, None),
+        StrptimeOptions {
+            format: Some("%Y-%m-%dT%H:%M:%SZ".to_string()),
+            strict: false,
+            exact: true,
+            cache: true,
+        },
+        lit("null"),
+    ));
+
+    println!("{:?}", lf.collect());
+
+    Ok(())
+}
+
+#[test]
+fn test_strptime_time() -> Fallible<()> {
+    let lf = df!("t" => &["+262142-01-01T01:02:03.232345Z"])?
+        .lazy()
+        .with_column(col("t").str().strptime(
+            DataType::Time,
+            StrptimeOptions {
+                format: Some("%Y-%m-%dT%H:%M:%S.fZ".to_string()),
+                strict: false,
+                exact: true,
+                cache: true,
+            },
+            lit("null"),
+        ));
+
+    let expected = DataFrame::new(vec![Series::new("t", vec![None::<NaiveTime>])])?;
+    assert_eq!(lf.collect()?, expected);
```

Comment:
If I'm reading this correctly, there's no data in the expected frame. Is this because there's a problem with the decimal seconds? This behavior is really different from the integer seconds that follow. Consider splitting this into two tests, and either in the test name or a comment, explain why this one is different.